src/test/FreeC/Pass/KindCheckPassTests.hs

```diff
@@ -26,26 +26,26 @@ testValidTypes = context "valid types" $ do
   it "should accept constant type constructors" $ do
     input <- expectParseTestType "()"
     shouldSucceed $ do
-      _ <- defineTestTypeCon "()" 0
+      _ <- defineTestTypeCon "()" 0 ["()"]
       checkType input
   it "should accept constant type synonyms" $ do
     input <- expectParseTestType "Name"
     shouldSucceed $ do
       _ <- defineTestTypeSyn "Name" [] "String"
-      _ <- defineTestTypeCon "String" 0
+      _ <- defineTestTypeCon "String" 0 []
       checkType input
   it "should accept fully applied type constructors" $ do
     input <- expectParseTestType "State Int Int"
     shouldSucceed $ do
-      _ <- defineTestTypeCon "State" 2
-      _ <- defineTestTypeCon "Int" 0
+      _ <- defineTestTypeCon "State" 2 []
+      _ <- defineTestTypeCon "Int" 0 []
       checkType input
   it "should accept fully applied type synonyms" $ do
     input <- expectParseTestType "State Int Int"
     shouldSucceed $ do
       _ <- defineTestTypeSyn "State" ["s", "a"] "s -> (,) s a"
-      _ <- defineTestTypeCon "Int" 0
-      _ <- defineTestTypeCon "(,)" 2
+      _ <- defineTestTypeCon "Int" 0 []
+      _ <- defineTestTypeCon "(,)" 2 []
```

Comment:
If the last argument resembles the constructor names, it shouldn't be empty for a pair, right?

Reply:
In this test case we do not care whether the data type has constructors or not. It is just important that there is a binary type constructor (which was called `(,)` in this case but must not be confused for the pair data type from Haskell). If the names of the constructors were provided, additional environment entries for the constructors would also have to be specified.

Reply:
So it's a bit of a hack to use this function in such a way, because, originally, it was supposed to be used differently? ; )

Reply:
I wouldn't call it a hack. Since the environment in test cases should only contain the information that is necessary for the test, the code below implies that it is important that the binary data type has a data constructor. 

```haskell
_ <- defineTestTypeCon "(,)" 2 ["(,)"]
_ <- defineTestCon "(,)" 2 "forall a b. a -> b -> (,) a b"
```

But that is not the case. The additional `defineTestCon` just distracts from the `defineTestTypeSyn`. Thus, I think it is right to specify no constructors. I understand that the use of well known names such as `(,)` may be confusing at first. However, names such as `Foo` and `Bar` don't communicate the arity of the type constructors at all, but the arity (especially of `State`) is very important for the test cases.